File: Logica de Programação/Subalgoritmo/SaudaHora.java
import java.util.Scanner; // Importa Scanner para entrada de dados

public class SaudaHora {

    // Programa Principal
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        int hora;

        System.out.print("Digite seu Horario (Ex: 12 para 12 horas): ");
        hora = entrada.nextInt(); // Lê a hora informada pelo usuário

        saudacao(hora); // Chama o procedimento passando a hora como parâmetro

        entrada.close(); // Fecha o Scanner após uso
    }

    // Procedimento que exibe uma saudação baseada no horário informado
    public static void saudacao(int hora) {
        String msg;

        if (hora < 6)
            msg = "Boa Madrugada!"; // Horário antes das 6h
        else if (hora < 12)
            msg = "Bom Dia!"; // Horário entre 6h e 11h59
        else if (hora < 18)
            msg = "Boa Tarde!"; // Horário entre 12h e 17h59
        else if (hora < 24)
            msg = "Boa Noite!"; // Horário entre 18h e 23h59
        else
            msg = "Horario Indisponível."; // Qualquer valor inválido

        // Exibe a saudação concatenada com a mensagem de boas-vindas
        System.out.println(msg + " Seja Bem-Vindo a FIAP!");
    }
}

File: Logica de Programação/Subalgoritmo/SaudaHora.py
# Método 1: função que retorna a saudação
def saudacao(hora):
    if hora < 6:
       return "Boa Madrugada!"
    elif hora < 12:
       return "Bom dia!"
    elif hora < 18:
       return "Boa Tarde!"
    elif hora < 24:
       return "Boa Noite!"
    else:
        return "Horário Indisponível."

hora = int(input("Digite seu horário (Ex 12 para 12 horas): "))
# Exibe a mensagem retornada pela função junto com saudação fixa
print(f"{saudacao(hora)} Seja Bem-Vindo a FIAP!")



# Método 2: função que já imprime a saudação diretamente
def saudacao2(hora):
    if hora < 6:
       msg = "Boa Madrugada!"
    elif hora < 12:
       msg = "Bom dia!"
    elif hora < 18:
       msg = "Boa Tarde!"
    elif hora < 24:
       msg = "Boa Noite!"
    else:
       msg = "Horário Indisponível."

    print(f"{msg} Seja Bem-Vindo a FIAP!!!")
    
# Chama a função que imprime direto
saudacao2(hora)
